consts/src/lib.rs

```diff
@@ -412,39 +412,38 @@ mod edhoc_parser {
         let mut ead_value = None::<EdhocMessageBuffer>;
 
         // assuming label is a single byte integer (negative or positive)
-        let label = match buffer.get(0) {
-            Some(b) => *b,
-            _ => return Err(EDHOCError::ParsingError),
-        };
-
-        let (label, is_critical) = if CBORDecoder::is_u8(label) {
-            // CBOR unsigned integer (0..=23)
-            (label as u8, false)
-        } else if CBORDecoder::is_i8(label) {
-            // CBOR negative integer (-1..=-24)
-            (label - (CBOR_NEG_INT_1BYTE_START - 1), true)
-        } else {
-            return Err(EDHOCError::ParsingError);
-        };
-
-        if buffer.len() > 1 {
-            // EAD value is present
-            let slice = match buffer.get(1..buffer.len()) {
-                Some(slice) => slice,
-                _ => return Err(EDHOCError::ParsingError),
+        if let Some(b) = buffer.get(0) {
+            let label = *b;
+            let res = if CBORDecoder::is_u8(label) {
+                // CBOR unsigned integer (0..=23)
+                Ok((label as u8, false))
+            } else if CBORDecoder::is_i8(*b) {
+                // CBOR negative integer (-1..=-24)
+                Ok((label - (CBOR_NEG_INT_1BYTE_START - 1), true))
+            } else {
+                Err(EDHOCError::ParsingError)
             };
-
-            let mut buffer = EdhocMessageBuffer::new();
-            buffer.content[..slice.len()].copy_from_slice(slice);
-            buffer.len = slice.len();
-            ead_value = Some(buffer);
-        }
-        ead_item = Some(EADItem {
-            label,
-            is_critical,
-            value: ead_value,
-        });
-        Ok(ead_item)
+            if let Ok((label, is_critical)) = res {
+                if buffer.len() > 1 {
+                    // EAD value is present
+                    let slice = &buffer[1..buffer.len()];
```

Comment:
This is replacing an explicit panic from `.get()` with an implicit one.

If you want to cut down panic options, could replace the `[1..buffer.len()]` with `[1..]` as a start, but you can also take it further and start with
```rust
if let Ok((&first, tail)) = buffer.split_first() {
    /* use first */
    /* no need for buffer.len() > 1, but go right with: */
    if Ok((&label, slice)) = tail.split_first() {
        /* use CBOR decoder on label or err out */
        /* slice here contains exactly what the annotated line panicingly  */
    }
} else {
    Err(...)
}
```

Reply:
Yes, I thought of `[1..]` on the way back home 🙃, but not the second one. Updated.